crates/burn-cube/src/lib.rs

```diff
@@ -54,6 +54,28 @@ pub fn calculate_cube_count_elemwise(num_elems: usize, cube_dim: usize) -> CubeC
     CubeCount::new(cube_count_x as u32, cube_count_y as u32, 1)
 }
 
+pub fn tensor_vectorization_factor(factors: &[u8], shape: &[usize], strides: &[usize]) -> u8 {
```

Comment:
It should be possible to vectorize over the dimension whose stride is 1, whatever dimension it is.